FastAPI wrapper fails for requests from a unix socket.

**Describe the bug**
We use a setup with a Gunicorn/FastAPI application binding to a unix socket. The FastAPI middleware from autodynatrace gives a `TypeError: 'NoneType' object is not subscriptable` on the `scope["server"]` variable.

```
Traceback (most recent call last):
  File "/oracle_sdi/venv/lib64/python3.11/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/oracle_sdi/venv/lib64/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/oracle_sdi/venv/lib64/python3.11/site-packages/fastapi/applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
  File "/oracle_sdi/venv/lib64/python3.11/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/oracle_sdi/venv/lib64/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/oracle_sdi/venv/lib64/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/oracle_sdi/venv/lib64/python3.11/site-packages/autodynatrace/wrappers/fastapi/middleware.py", line 26, in __call__
    host = "{}:{}".format(scope["server"][0], scope["server"][1])
                          ~~~~~~~~~~~~~~~^^^
TypeError: 'NoneType' object is not subscriptable
```

**To Reproduce**
Steps to reproduce the behavior:
1. Configure Nginx to pass the requests to a unix socket
2. Configure Gunicorn to bind to unix socket
3. Configure Gunicorn with autodynatrace wrapper
4. Start Gunicorn with a FastAPI application
5. Send a request to Nginx
6. Gunicorn will throw an Exception: `[ERROR] Exception in ASGI application`

**Expected behavior**
According to the  ASGI specifications for http connection scope, the information on "server" is optional (https://asgi.readthedocs.io/en/latest/specs/www.html). However the implementation of the middleware wrapper for FastAPI (wrappers/fastapi/middleware.py) assumes the `[host, port]` iterable unconditionally. 

Since the specification allows a `None` value and two-item iterable `[path, None]` too, these two cases need to be handled as well.

Example fix:
```
        host = ""
        # https://asgi.readthedocs.io/en/latest/specs/www.html
        # Optional; if missing defaults to None.
        if scope.get("server"):
            if scope["server"][1]:
                # [host:port]
                host = "{}:{}".format(scope["server"][0], scope["server"][1])
            else:
                # [path, None]
                host = "{}".format(scope["server"][0])
```

**Server:**
 - OS: Red Hat Enterprise Linux release 9.4 (Plow)
 - Python 3.11.7
 - autodynatrace 2.1.1

--- a/autodynatrace/wrappers/fastapi/middleware.py
+++ b/autodynatrace/wrappers/fastapi/middleware.py
@@ -1,7 +1,7 @@
 import os
 
-from ...sdk import sdk
 from ...log import logger
+from ...sdk import sdk
 
 
 class DynatraceASGIMiddleware:
@@ -9,7 +9,6 @@ class DynatraceASGIMiddleware:
         self.app = app
 
     async def __call__(self, scope, receive, send):
-
         # If this is not an http request, do nothing and return
         if scope["type"] != "http":
             return await self.app(scope, receive, send)
@@ -23,20 +22,31 @@ class DynatraceASGIMiddleware:
 
         dt_tag = headers.get("x-dynatrace")
 
-        host = "{}:{}".format(scope["server"][0], scope["server"][1])
-        url = "{}://{}{}?{}".format(scope["scheme"], host, scope["path"], scope["query_string"].decode())
+        host = ""
+        # scope["server"] is optional, if missing, it defaults to None.
+        # https://asgi.readthedocs.io/en/latest/specs/www.html
+        if scope.get("server") and scope["server"][1]:
+            # [host, port]
+            host = f"{scope['server'][0]}:{scope['server'][1]}"
+        elif scope.get("server"):
+            # [path, None]
+            host = f"{scope['server'][0]}"
+
+        url = f"{scope['scheme']}://{host}{scope['path']}?{scope['query_string'].decode()}"
         method = scope.get("method", "GET")
         app = scope.get("app")
         app_name = app.title if app is not None else "FastAPI"
         root = scope.get("root_path", "/")
 
-        virtual_host = os.environ.get("AUTODYNATRACE_VIRTUAL_HOST", "{}".format(host))
-        app_name = os.environ.get("AUTODYNATRACE_APPLICATION_ID", "{}".format(app_name))
+        virtual_host = os.environ.get("AUTODYNATRACE_VIRTUAL_HOST", f"{host}")
+        app_name = os.environ.get("AUTODYNATRACE_APPLICATION_ID", f"{app_name}")
         context_root = os.environ.get("AUTODYNATRACE_CONTEXT_ROOT", root or "/")
 
         with sdk.create_web_application_info(virtual_host, app_name, context_root) as web_app_info:
-            with sdk.trace_incoming_web_request(web_app_info, url, method, headers=headers, str_tag=dt_tag) as tracer:
-                logger.debug("dynatrace - asgi - Tracing url: '{}' with tag {}".format(url, dt_tag))
+            with sdk.trace_incoming_web_request(
+                web_app_info, url, method, headers=headers, str_tag=dt_tag,
+            ) as tracer:
+                logger.debug(f"dynatrace - asgi - Tracing url: '{url}' with tag {dt_tag}")
 
                 async def wrapped_send(message):
                     if message.get("type") == "http.response.start" and "status" in message:

